client/resources/md5mesh.cpp

```diff
@@ -366,10 +365,10 @@ void MD5Mesh::UpdateIndexList() {
 		}
 		ObjectGroup* objgroup = this->meshes[i]->object_groups[0];
 		std::string material_name = int_mesh.shader;
-		material_name = material_name.substr(
-								material_name.find_last_of("/") + 1,
-								material_name.find_last_of(".") - material_name.find_last_of("/") - 1)
-						+ "_material";
+		size_t slash = material_name.find_last_of("/");
+		slash = (slash == std::string::npos) ? 0 : (slash + 1); // skip past any slash
+		size_t dot = material_name.find_last_of(".");
+		material_name = material_name.substr(slash, dot - slash) + "_material";
```

Comment:
This could be a nice utility function to get the filename since I see it is used in obj.cpp as well.

Reply:
I started to add this as a method of FilePath because it makes more sense that way, but that will have to be part of another PR, I kind of ended up dealing with other problems in the filesystem code.